client/rest-high-level/src/main/java/org/elasticsearch/client/security/DeletePrivilegesRequest.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.security;
+
+import org.elasticsearch.client.Validatable;
+
+import java.util.Objects;
+
+/**
+ * A request to delete application privileges
+ */
+public final class DeletePrivilegesRequest implements Validatable {
+
+    private final String application;
+    private final String[] privileges;
+    private final RefreshPolicy refreshPolicy;
+
+    public DeletePrivilegesRequest(String application, String... privileges) {
+        this(application, privileges, RefreshPolicy.IMMEDIATE);
+    }
+
+    public DeletePrivilegesRequest(String application, String[] privileges, RefreshPolicy refreshPolicy) {
+        this.application = Objects.requireNonNull(application, "application is required");
+        this.privileges = Objects.requireNonNull(privileges, "privileges are required");
+        this.refreshPolicy =  Objects.requireNonNull(refreshPolicy, "refresh policy is required");
```

Comment:
In other APIs, we have allowed null and used `RefreshPolicy.getDefault()`. I don't have strong feelings for implicit vs explicit in this case but I think consistency helps. Would it make sense to make a decision and use this for all APIs? @hub-cap WDYT ?

Reply:
I made it consistent and changed the code to use the `RefreshPolicy.getDefault()`.

We can still change this if someone thinks it should be immediate too.